libr/anal/esil.c

```diff
@@ -3169,8 +3258,10 @@ static void r_anal_esil_setup_ops(RAnalEsil *esil) {
 	OP ("--", esil_dec, 1, 1, OT_MATH);
 	OP ("--=", esil_deceq, 0, 1, OT_MATH | OT_REGW);
 	OP ("/", esil_div, 1, 2, OT_MATH);
+	OP ("~/", esil_signed_div, 1, 2, OT_MATH);
```

Comment:
we need signed multiplication, addition and substraction, i guess @condret would prefer a SIGN operator instead of adding so many new instructions to the esil vm.

Reply:
After sign extension multiplication, addition, subtraction are all the same as unsigned, and yea I also added the signed division operator (~/) and signed modulus (~%). Also ~ (SIGN) pops two values from the stack, the value to sign extend and the size of the value before extension, and it pushes the sign extended 64 bit value to the stack. I could modify it to have a target bit size also but I haven't found a place where that would be necessary.

Reply:
I also want to try to minimize the number of operators, I could possibly eliminate the need for the signed mod but it would make expressions for instrs like IDIV even wilder than they are already. I think otherwise they are the only new operators required to cover all simple math with signed integers.